cfn-resources/stream-instance/mongodb-atlas-streaminstance.json

```diff
@@ -0,0 +1,245 @@
+        },
+        "Protocol": {
+          "type": "string",
+          "description": "Describes the transport type. Can be either PLAINTEXT or SSL."
+        }
+      },
+      "additionalProperties": false
+    },
+    "StreamsConnection": {
+      "type": "object",
+      "description": "Settings that define a connection to an external data store.",
+      "properties": {
+        "Name": {
+          "type": "string",
+          "description": "Human-readable label that identifies the stream connection."
+        },
+        "Type": {
+          "type": "string",
+          "description": "Type of the connection. Can be either Cluster or Kafka.",
+          "enum": [
+            "Kafka",
+            "Cluster",
+            "Sample"
+          ]
+        },
+        "Authentication": {
+          "$ref": "#/definitions/StreamsKafkaAuthentication"
+        },
+        "BootstrapServers": {
+          "type": "string",
+          "description": "Comma separated list of server addresses."
+        },
+        "Security": {
+          "$ref": "#/definitions/StreamsKafkaSecurity"
+        },
+        "ClusterName": {
+          "type": "string",
+          "description": "Name of the cluster configured for this connection."
+        },
+        "DbRoleToExecute": {
+          "$ref": "#/definitions/DBRoleToExecute"
+        }
+      },
+      "additionalProperties": false
+    }
+  },
+  "properties": {
+    "Profile": {
+      "type": "string",
+      "description": "The profile is defined in AWS Secret manager. See [Secret Manager Profile setup](../../../examples/profile-secret.yaml).",
+      "default": "default"
+    },
+    "Name": {
+      "description": "Human-readable label that identifies the stream connection.",
+      "type": "string"
+    },
+    "DataProcessRegion": {
+      "$ref": "#/definitions/StreamsDataProcessRegion"
+    },
+    "StreamConfig": {
+      "$ref": "#/definitions/StreamConfig"
+    },
+    "Id": {
+      "description": "Unique 24-hexadecimal character string that identifies the project.",
+      "type": "string"
+    },
+    "Connections": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/StreamsConnection"
+      },
+      "insertionOrder": false
+    },
+    "GroupId": {
+      "description": "Unique 24-hexadecimal character string that identifies the project.",
+      "type": "string"
+    },
+    "Hostnames": {
+      "description": "List that contains the hostnames assigned to the stream instance.",
+      "type": "array",
+      "items": {
+        "type": "string"
+      },
+      "insertionOrder": false
+    }
+  },
+  "additionalProperties": false,
+  "required": [
+    "GroupId",
+    "Name",
+    "DataProcessRegion",
+    "StreamConfig"
+  ],
+  "readOnlyProperties": [
+    "/properties/Id",
+    "/properties/Connections",
+    "/properties/Hostnames"
+  ],
+  "primaryIdentifier": [
+    "/properties/Name",
+    "/properties/Profile"
+  ],
```

Comment:
Would the primaryIdentifier be compose of the name + project id? Also curious why we are adding profile as part of the primary identified in all our resources, but that is a broader conversation.

Reply:
@maastha probably has more context on why Profile is in primaryIdentifiers in all resources. 
On another note, I added projectId in the primaryIdentifier

Reply:
looping in @andreaangiolillo, The main reason was that only primary identifier fields can be accessed during the read operation. The profile was a workaround to avoid defining public/private/projectid as primary identifier parameters for every resource